sync/base.py

```diff
@@ -636,7 +636,7 @@ def load_by_status(cls, repo, subtype, status):
         return rv
 
     def _save(self, data: dict[str, Any], message: str,
-              commit_builder: CommitBuilder = None) -> Any | None:
+              commit_builder: CommitBuilder | None = None) -> Any | None:
```

Comment:
I see we're using a mix of `Optional[Foo]` and `Foo | None`. I somewhat prefer the first rather than inlining variants and we should probably pick one, but don't have to do that now.

Reply:
I tried to stick to `Foo | None` since I've seen it's used a lot in the codebase (but I personally also prefer `Optional[Foo]`). But in that one place, where I used `Optional[Foo]`, there was some syntax problem with `Foo | None`.

Reply:
OK, let's defer this to a followup issue.